[dynamo] Record nn_module_stack also for unspecialized nn modules.

zhxchen17 opened the pull request:
Summary: Currently node metadata "nn_module_stack" is only being used by export. For some export model, we still want to retain nn_module_stack for unspecialized module for various purposes. This diff add a path to also record nn_module_stack when unspecialized module has a source available.

Test Plan: test_export_nn_module_stack_patched_module

Differential Revision: D48841193



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305

yanboliang commented on torch/_dynamo/variables/nn_module.py:
```diff
@@ -686,9 +687,15 @@ def call_function(
+        with ctx:
+            return variables.UserFunctionVariable(
+                fn, source=source, **options
+            ).call_function(tx, [self] + list(args), kwargs)
```
Can add a dynamo test in ```dynamo/test_module.py``` to check if it works well when this inlining failed. Otherwise, this looks good!

zhxchen17 replied:
np, will do that!

zhxchen17 commented:
@pytorchbot merge

zhxchen17 committed:
[dynamo] Record nn_module_stack also for unspecialized nn modules. (#108281)

Summary:
Currently node metadata "nn_module_stack" is only being used by export. For some export model, we still want to retain nn_module_stack for unspecialized module for various purposes. This diff add a path to also record nn_module_stack when unspecialized module has a source available.


Test Plan:
test_export_nn_module_stack_patched_module




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng anijain2305

Reviewed By: yanboliang, tugsbayasgalan

Differential Revision: D48841193

Pulled By: zhxchen17

zhxchen17 commented:
@pytorchbot merge -f "landing has been delayed for 1 day"
